scripts/link-checker.js

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env node
+/*
+Copyright (c) 2018-2020 Uber Technologies, Inc.
+
+This source code is licensed under the MIT license found in the
+LICENSE file in the root directory of this source tree.
+*/
+
+/* eslint-env node */
+/* eslint-disable flowtype/require-valid-file-annotation */
+
+const childProcess = require('child_process');
+const fetch = require('isomorphic-fetch');
+
+const {
+  BUILDKITE_COMMIT,
+  BUILDKITE_BRANCH,
+  BUILDKITE_PULL_REQUEST_REPO,
+  ZEIT_NOW_TOKEN,
+  ZEIT_NOW_TEAM_ID,
+  ZEIT_NOW_PROJECT_ID,
+} = process.env;
+
+main().catch(er => {
+  console.error(er);
+  process.exit(1);
+});
+
+async function main() {
+  console.log('Checking documentation site links');
+  console.log('Resolving deployment URL');
+
+  let url = await getDeploymentURL();
+
+  console.log(`Resolved URL to [${url}]`);
+
+  let attemptCounter = 0;
+  let maxAttempts = 35;
+  let urlExists = false;
+
+  process.stdout.write('Pinging URL...');
+  do {
+    attemptCounter += 1;
+    try {
+      await fetch(url);
+      urlExists = true;
+    } catch (er) {
+      process.stdout.write('.');
+      await sleep(30000);
+    }
+  } while (!urlExists && attemptCounter < maxAttempts);
+
+  if (urlExists) {
+    console.log(' Connected!');
+    console.log('Running link checker');
+    childProcess.execSync(
+      `yarn blc ${url} -ro --exclude components/avatar --exclude example.com --exclude github.com`,
+      {stdio: 'inherit'},
+    );
+  } else {
+    console.log('');
+    throw new Error('Maximum number of attempts reached.');
+  }
+}
+
+async function getDeploymentURL() {
```

Comment:
```suggestion
export async function getDeploymentURL() {
```

I'd like to reuse this for the documentation site vrt story

Reply:
ops, it won't work this way - can you split this out to a module, so it can be reused?

Reply:
or we can simply merge as is, and happy to do that in my pr

Reply:
what is your idea for that?

Reply:
I'd like to get the zeit deployment url to create a snapshot of the ProgressStep component (that seems to be one that doesn't change a lot), and one one a blogpost

Reply:
I think running VRT tests across the network may be asking for bugs and heartache

Reply:
That's a good call-out - I think another option we have is to build the doc site in the CI workflow. Previously we've removed that because it slowed down the pipeline considerably. 

I guess we have two options here

* try out and see if over-the-network puppeteer screenshots would be flaky
* try out building the doc site locally again

which one would you try first? did I miss a third one?

Reply:
Is the requirement: VRT tests must be run on examples published to the docs site?

Could docs site builds be avoided by including the examples into the storybook/e2e server?